arches/app/models/resource.py

```diff
@@ -190,15 +184,36 @@ def bulk_save(resources, flat=False):
             tiles = [tile for resource in resources for tile in resource.tiles]
 
         # need to save the models first before getting the documents for index
-        Resource.objects.bulk_create(resources)
-        TileModel.objects.bulk_create(tiles)
+        # need to handle if the bulk load is appending tiles to existing resources/
+        existing_resources_ids = Resource.objects.filter(
+            resourceinstanceid__in=[resource.resourceinstanceid for resource in resources]
+        ).values_list("resourceinstanceid", flat=True)
 
-        for resource in resources:
-            resource.save_edit(edit_type="create")
+        resources_to_update = [resource for resource in resources if resource.resourceinstanceid in existing_resources_ids]
+        resources_to_create = [resource for resource in resources if resource.resourceinstanceid not in existing_resources_ids]
 
-        resources[0].tiles[0].save_edit(note=f"Bulk created: {len(tiles)} for {len(resources)} resources.", edit_type="bulk_create")
+
+        Resource.objects.bulk_create(resources_to_create)
+        TileModel.objects.bulk_create(tiles)
```

Comment:
@whatisgalen in the local code we are using we found that we needed to include the following to ensure that actions in the pre_tile_save were actioned...
```py
# need to run logic in pre_tile_save fro the datatypes otherwise resource relationships and file aren't loaded correctly.
# Needs to be some kind of bulk_pre_tile_save
    for tile in tiles:
        for nodeid in tile.data.keys():
            try:
                node_datatype = datatype_factory.get_instance(node_datatypes[nodeid])
                node_datatype.pre_tile_save(tile, nodeid)
            except:
                pass
```